### ISSUE TITLE
Initial syncing should not save empty blob

### ISSUE THREAD

**Issue by terencechain** [prysmaticlabs/prysm#12796]
```
[2023-08-24 07:43:56]  INFO blockchain: Finished applying state transition attestations=1 payloadHash=0x1bb58292fe6c slot=58817 syncBitsCount=512 txCount=0
[2023-08-24 07:43:56]  WARN initial-sync: Failed to save blob sidecar error=nil or empty blob sidecars
```

**Comment by kasey** [comment]
Is the reproduction case for this to sync across slot 58817 on devnet 8?

**Comment by kasey** [comment]
PRs fixing this issue are 
https://github.com/prysmaticlabs/prysm/pull/12797
and
https://github.com/prysmaticlabs/prysm/pull/12798
these both target `devnet8-fix`. We'll keep this issue open until those changes are cherry-picked to deneb-integration.

**Comment by kasey** [comment]
resolved by #12798
